components/requireAuth.tsx

```diff
@@ -17,13 +21,65 @@ export default function requireAuth(ChildComponent, authRole) {
       if (!authUser || !authUser.email) router.push('/signin')
       else {
         if (authUser.role !== authRole) router.push('/404')
-        else setPageReady(true)
+        else {
+          if (authRole === Roles.APPLICANT) {
+            if (auth.currentUser.emailVerified) setPageReady(true)
+            else
+              setError(
+                'Please verify your email first, if verified try to reload/signing out.',
+              )
+          } else setPageReady(true)
+        }
       }
     }, [loading, authUser])
 
     return !pageReady ? (
       <MainLayout>
-        <Loading message="Loading your page!" />
+        {!error && !emailSent ? (
+          <Loading message="Loading your page!" />
+        ) : (
+          <div className="flex flex-col justify-center items-center my-24">
+            {!error ? null : (
+              <div className="flex justify-center items-center">
+                <p className="text-3xl text-red-850 font-black">Error: </p>
+                <p className="text-3xl text-red-850 ml-5">{error}</p>
+              </div>
+            )}
+            {!emailSent ? null : (
+              <div className="flex justify-center items-center">
+                <p className="text-3xl text-blue-850 text-center">
+                  A new verification email is sent. If not recieved, try again
+                  later!
+                  <br /> If verified, try reloaing!
+                </p>
+              </div>
+            )}
+            <div className="flex justify-center items-center mt-10">
+              <button
+                className="text-white text-base md:text-lg bg-blue-850 py-2 px-4 rounded-3xl"
+                onClick={() => {
+                  if (!emailSent) {
```

Comment:
@himanshuvarandani It's possible to spam-hit the email button, which should be fixed.
Since `emailSent` doesn't persist on reloads, we could fetch it's value from Firestore to add rate limiting.
Can store epoch in Firestore to compare time delta. Emails could be sent again after `n` days (n=1 for simplicity).